clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -86,9 +87,34 @@ def get_unschedulable_pods(self) -> List[KubernetesPod]:
                 unschedulable_pods.append(pod)
         return unschedulable_pods
 
+    def _pod_matches_node_selector_or_affinity(self, pod: KubernetesPod) -> bool:
+        if pod.spec.node_selector:
+            for key, value in pod.spec.node_selector.items():
+                if key == self.pool_label_key:
+                    if value == self.pool:
+                        return True
+                    else:
+                        return False
+
+        selector_requirement = V1NodeSelectorRequirement(
+            key=self.pool_label_key, operator='In', values=[self.pool]
+        )
+        if pod.spec.affinity and pod.spec.affinity.node_affinity:
+            node_affinity = pod.spec.affinity.node_affinity
+            if node_affinity.required_during_scheduling_ignored_during_execution:
```

Comment:
we're going to have to come back to this at some point and add support for `required_during_scheduling_required_during_execution`. I'm also wondering if we should support `preferredDuringSchedulingIgnoredDuringExecution` too?  see explanation here https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#node-affinity. maybe we should just do the latter now and leave a TODO/ticket to pickup the required_during_scheduling_required_during_execution work when it's been added

Reply:
As per our last conversation, I have included support for preferredDuringSchedulingIgnoredDuringExecution.